### PR TITLE
IPDK Negative Test Cases

### PR DESCRIPTION
**Pull request by sjwahub**
Unit Test log  [ptf_tests_run_neg_final_1.log](https://github.com/ipdk-io/ptf-tests/files/10789977/ptf_tests_run_neg_final_1.log)

### CODE REVIEW

File: common/lib/port_config.py [hunk 1]
```diff
@@ -67,6 +67,21 @@ def gnmi_ctl_set(self, params):
                 raise ExecuteCMDException(f"Failed to execute command {cmd}")
             log.passed(f"{cmd}")
             return output
+        
+        def gnmi_ctl_set_invalid_args(self, params):
```
Comment (dkaranjai) [inline common/lib/port_config.py]: you dont need this. this is the reverse of the existing one. the utils function is sufficient with the pass/fail checks reversed.

File: common/utils/gnmi_ctl_utils.py [hunk 2]
```diff
@@ -40,6 +40,21 @@ def gnmi_set_params(params):
 
     return output
 
+def gnmi_ctl_set_and_verify_invalid_args(params):
```
Comment (dkaranjai) [inline common/utils/gnmi_ctl_utils.py:44]: refer to the comment on port_config.py

File: tests/LNT_ECMP_2VM_2Host_Invalid_Value.py [hunk 3]
```diff
@@ -0,0 +1,499 @@
...
+
+        # Configuring VMs
+        for i in range(len(self.conn_obj_list)):
+            log.info("Configuring VM....")
+            test_utils.configure_vm(self.conn_obj_list[i], vm_cmd_list[i])
+
+        # Bring up TAP ports
+        if not gnmi_ctl_utils.ip_set_dev_up(self.tap_port_list[0]):
+            self.result.addFailure(self, sys.exc_info())
+            self.fail("Failed to bring up {self.tap_port_list[0]}")
```
Comment (dkaranjai) [inline tests/LNT_ECMP_2VM_2Host_Invalid_Value.py]: f missing

File: tests/LNT_ECMP_2VM_2Host_Invalid_Value.py [hunk 4]
```diff
@@ -0,0 +1,499 @@
...
+            nexthop_list,
+            device_list,
+            weight_list,
+            remote=True,
+            hostname=self.config_data["client_hostname"],
+            username=self.config_data["client_username"],
+            password=self.config_data["client_password"],
+        ):
+            self.result.addFailure(self, sys.exc_info())
+            self.fail(f"Failed to add route")
```
Comment (dkaranjai) [inline tests/LNT_ECMP_2VM_2Host_Invalid_Value.py]: f not required

File: tests/LNT_ECMP_2VM_2Host_Invalid_Value.py [hunk 5]
```diff
@@ -0,0 +1,499 @@
...
+        log.info(f"Delete ip route")
+        if not gnmi_ctl_utils.iproute_del(
+            self.config_data["vxlan"]["tep_ip"][1].split("/")[0]
+        ):
+            self.result.addFailure(self, sys.exc_info())
+            self.fail(
+                f"Failed to delete route to {self.config_data['vxlan']['tep_ip'][1].split('/')[0]}"
+            )
+
+        log.info("Unconfiguration on remote host")
```
Comment (dkaranjai) [inline tests/LNT_ECMP_2VM_2Host_Invalid_Value.py]: replace unconfiguration with cleanup

File: tests/LNT_ECMP_2VM_2Host_Negative.py [hunk 6]
```diff
@@ -0,0 +1,719 @@
...
+        log.info(f"Delete ip route")
+        if not gnmi_ctl_utils.iproute_del(
+            self.config_data["vxlan"]["tep_ip"][1].split("/")[0]
+        ):
+            self.result.addFailure(self, sys.exc_info())
+            self.fail(
+                f"Failed to delete route to {self.config_data['vxlan']['tep_ip'][1].split('/')[0]}"
+            )
+
+        log.info("Unconfiguration on remote host")
```
Comment (dkaranjai) [inline tests/LNT_ECMP_2VM_2Host_Negative.py:637]: refer previous test script

**Review by dkaranjai — CHANGES_REQUESTED** [review]

**Comment by sjwahub** [comment]
Unit test log for code changes based on comment [ptf_tests_run_negaive_2.log](https://github.com/ipdk-io/ptf-tests/files/10793063/ptf_tests_run_negaive_2.log)

**Review by dkaranjai — APPROVED** [review]

**Pull request merged**